### PR TITLE
Refac: skipped BatchExecutionStateDomainService tests

### PR DESCRIPTION
**Pull request by johevemi**
## Description

This PR makes moves MapState in BatchExecutionStateDomainService to its own mapper class to be able to test their functionality in unit tests and removes some of the existing integration tests which basically test the same thing but with a lot more complexity

## References

* #782

### CODE REVIEW

**Review by BjarkeMeier — COMMENTED** [review]
IMHO, whenever you start making private methods public in order to test them it calls for a reflection on whether this is an indication that things should have less coupling. Normally I would consider refactoring to a delegation pattern. In this case I would introduce something like a `IBatchStateMapper`.

Side note: Is our model correct when a batch state has a bijective relation to a calculation state? (You don't have to answer 😄)

File: source/dotnet/Services/ProcessManager.IntegrationTests/BatchApplicationServiceTests.cs [hunk 1]
```diff
@@ -56,99 +56,84 @@ public BatchApplicationServiceTests(ProcessManagerDatabaseFixture processManager
...
 
-    [Fact(Skip = "Split into multiple tests when concepts are ready")]
-    public async Task When_RunIsPending_Then_BatchIsPending()
+    [Theory]
+    [InlineAutoMoqData(CalculationState.Pending, BatchExecutionState.Pending)]
+    [InlineAutoMoqData(CalculationState.Running, BatchExecutionState.Executing)]
+    [InlineAutoMoqData(CalculationState.Completed, BatchExecutionState.Completed)]
+    [InlineAutoMoqData(CalculationState.Canceled, BatchExecutionState.Canceled)]
+    [InlineAutoMoqData(CalculationState.Failed, BatchExecutionState.Failed)]
+    public void MapStateSuccess(CalculationState calculationState, BatchExecutionState expectedBatchExecutionState, BatchExecutionStateDomainService sut)
```
Comment (BjarkeMeier) [inline source/dotnet/Services/ProcessManager.IntegrationTests/BatchApplicationServiceTests.cs:65]: Can we find a better name for the test? How about `Map_ReturnsExpectedState`?

[the remark this answers is not in the document]
Comment (johevemi) [reply source/dotnet/Services/ProcessManager.IntegrationTests/BatchApplicationServiceTests.cs]: moved to unittest project

Comment (johevemi) [reply source/dotnet/Services/ProcessManager.IntegrationTests/BatchApplicationServiceTests.cs]: Renamed the test to:
MapState_CalculationState_ExpectedBatchExecutionState
MapState_UnexpectedCalculationState_ThrowsArgumentOutOfRangeException
HandleNewState_BatchExecutionState_ExpectedBatchExecutionState
HandleNewState_UnexpectedBatchExecutionState_ThrowsArgumentOutOfRangeException

**Comment by johevemi** [comment]
> IMHO, whenever you start making private methods public in order to test them it calls for a reflection on whether this is an indication that things should have less coupling. Normally I would consider refactoring to a delegation pattern. In this case I would introduce something like a `IBatchStateMapper`.
> 
> Side note: Is our model correct when a batch state has a bijective relation to a calculation state? (You don't have to answer 😄)

Is introducing an IBatchStateMapper necessary when the two methods are only used in BatchExecutionStateDomainService?

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**